fv/containers/containers.go

```diff
@@ -308,3 +312,84 @@ func StartSingleNodeEtcdTopology() (felix, etcd *Container, client client.Interf
 	success = true
 	return
 }
+
+// StartSingleNodeEtcdTopology starts an etcd container and a single Felix container; it initialises
+// the datastore and installs a Node resource for the Felix node.
+func StartTwoNodeEtcdIPIPTopology() (felixes []*Container, etcd *Container, client client.Interface) {
+	log.Info("Starting a single-node etcd topology.")
+	success := false
+	var err error
+	defer func() {
+		if !success {
+			log.WithError(err).Error("Failed to start topology, tearing down containers")
+			for _, felix := range felixes {
+				felix.Stop()
+			}
+			etcd.Stop()
+		}
+	}()
+
+	// First start etcd.
+	etcd = RunEtcd()
+
+	// Connect to etcd.
+	client = utils.GetEtcdClient(etcd.IP)
+	Eventually(func() error {
+		log.Info("Initializing the datastore...")
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		err = client.EnsureInitialized(
+			ctx,
+			"test-version",
+			"felix-fv",
+		)
+		log.WithError(err).Info("EnsureInitialized result")
+		if err != nil {
+			log.WithError(err).Warn("EnsureInitialized failed")
+			return err
+		}
+		ctx, cancel = context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		ipPool := api.NewIPPool()
+		ipPool.Name = "test-pool"
+		ipPool.Spec.CIDR = "10.65.0.0/16"
+		ipPool.Spec.IPIPMode = api.IPIPModeAlways
+		_, err = client.IPPools().Create(ctx, ipPool, options.SetOptions{})
+		return err
+	}).ShouldNot(HaveOccurred())
```

Comment:
I guess all that is shared with the single node setup case - can it be commonized?

Reply:
Or you could just make the number of felixes a parameter to this whole function, and use it in the single case too.